### PR TITLE
[WIP] feat(addon-editor): hide tools in dropdown after overflow

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Part of #1302

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/156157734-fb1ee1fd-4d3f-4f9f-a766-98ae77416cd7.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/addon-editor/components/toolbar-new/toolbar-new.template.html [hunk 1]
```diff
@@ -5,12 +5,305 @@
...
 
-<section
-    *ngIf="texts$ | async as texts"
-    tuiToolbarNavigationManager
-    class="t-tools-wrapper"
->
-    <div *ngIf="enabled(TuiEditorTool.Undo)" class="t-block">
+<div class="t-tools-panel">
+    <section tuiToolbarNavigationManager class="t-tools-wrapper">
+        <div
```
Comment (waterplea) [inline projects/addon-editor/components/toolbar-new/toolbar-new.template.html:10]: There's a lot of repetition here, you can probably `*ngFor` it somehow.

**Review by waterplea — COMMENTED** [review]
How will it work for custom tools?

File: projects/addon-editor/components/toolbar-new/toolbar-new.template.html [hunk 2]
```diff
@@ -5,12 +5,305 @@
...
+            *ngIf="enabled(TuiEditorTool.Undo)"
+            class="t-block"
+            [class.t-hidden]="toolsSetInDropdown.has(TuiEditorTool.Undo)"
+            (tuiToolbarIntersection)="changeVisible($event, TuiEditorTool.Undo)"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/toolbar-new/toolbar-new.template.html:14]: I tried to do this and it didn't work for me, while there are not many emit events, you can check this by executing console log

File: projects/addon-editor/components/toolbar-new/toolbar-new.template.html [hunk 3]
```diff
@@ -5,12 +5,305 @@
...
+    <ng-template #dropdownContent let-activeZone>
+        <div class="t-dropdown-tools">
+            <ng-container *ngIf="toolsSetInDropdown.has(TuiEditorTool.Size)">
+                <ng-container
+                    *ngTemplateOutlet="sizeTool; context: { $implicit: activeZone }"
+                ></ng-container>
+            </ng-container>
+
+            <ng-container
+                *ngIf="toolsSetInDropdown.has(TuiEditorGroupTools.Format)"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/toolbar-new/toolbar-new.template.html:204]: I don't understand, what do you mean (for example)?

File: projects/addon-editor/components/toolbar-new/toolbar-new.template.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/toolbar-new/toolbar-new.template.html:14]: not really, some blocks have some groups

image

**Comment by splincode** [comment]
@waterplea `How will it work for custom tools?`
this does not work for custom tools, because it is not clear how to extract templates

**Pull request closed without merging**